osu.Game/Online/API/APIAccess.cs

```diff
@@ -440,6 +440,13 @@ public void Logout()
             flushQueue();
         }
 
+        private void failFromAPIOffline(APIRequest req)
+        {
+            Debug.Assert(state.Value == APIState.Offline);
```

Comment:
This is not always the case though. `flushQueue` is also called in `Failing`.

Reply:
Yep, I'm fixing that now.

Reply:
I've split them back out and just adjusted the other message instead, since the state is not a given.